Sources/FirebaseCore/FirebaseOptions+Swift.swift

```diff
@@ -126,3 +126,30 @@ extension FirebaseOptions {
   @available(*, unavailable)
   public var appGroupId: String? { nil }
 }
+
+extension FirebaseOptions {
+  /// The format of a specified config file
+  public enum ConfigFormat {
+    /// Specifies that the file contents should be treated as a json string.
+    case json
+  }
+
+  /// An added initializer which will allow loading of a config from a specified file.
+  /// - Parameters:
+  ///   - path: The path where the config file can be found.
+  ///   - format: A format which describes how the content of the file should be treated.
+  public init?(_contentsOfFile path: String, format: ConfigFormat) {
+    switch format {
+      case .json:
+        do {
+          let config = try String(contentsOfFile: path)
```

Comment:
Hmm I disagree actually. Since we don't know what other loaders could need since they don't exist I don't think we can universally say that a `String` should be used for all of them. I could easily see some kind of Codable be used, a Data blob, or something else. I believe we should keep the string loading limited to the one loading function we know takes exactly a String.

Reply:
Hmm, interesting.  Yeah, I guess I see it as: everything is ultimately serialised to a file, which we load into memory, and then deserialise.  I think that this brings up an excellent point and why we have the difference of opinion, this code should be made more indirect:

```swift
let data: Data = Data(contentsOf: URL(fileURLWithPath: path), options: [.alwaysMapped])
```

Reply:
I can buy this as a better alternative, will fix